cpp/src/arrow/acero/unmaterialized_table.h

```diff
@@ -76,7 +76,7 @@ class UnmaterializedCompositeTable {
     if (Empty()) {
```

Comment:
We could try to make this file internal IMHO.

Reply:
Thank you. It always feels good to confirm something to be internal only. I'll rename them.

Reply:
Renamed.